gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -313,8 +314,10 @@ private void prepareHadoopJob(List<WorkUnit> workUnits) throws IOException {
 
     // Serialize source state to a file which will be picked up by the mappers
     Path jobStateFilePath = new Path(this.mrJobDir, JOB_STATE_FILE_NAME);
-    SerializationUtils.serializeState(this.fs, jobStateFilePath, this.jobContext.getJobState(), 100);
+    SerializationUtils.serializeState(this.fs, jobStateFilePath, this.jobContext.getJobState());
+    DistributedCache.addCacheFile(jobStateFilePath.toUri(), this.job.getConfiguration());
```

Comment:
We should use `MRJobLauncher.addHDFSFiles` which already does this, it also sets up a symlink to the file, this way the additional property `JOB_STATE_DISTRIBUTED_CACHE_NAME` should not longer be necessary.

Using the symlink should also simply the code added to the `setup` method.